torch/utils/cpp_extension.py

```diff
@@ -740,6 +764,24 @@ def _check_abi(self):
                    'Please set `DISTUTILS_USE_SDK=1` and try again.')
             raise UserWarning(msg)
 
+    def _check_cuda_version(self):
+        if CUDA_HOME:
+            nvcc = os.path.join(CUDA_HOME, 'bin', 'nvcc')
+            cuda_version_str = subprocess.check_output([nvcc, '--version']).strip().decode()
+            cuda_version = re.search(r'release (\d+[.]\d+)', cuda_version_str)
+            if cuda_version is not None:
+                cuda_str_version = cuda_version.group(1)
+                cuda_ver = parse_version(cuda_str_version)
```

Comment:
apparently this returns a type of `pkg_resources.extern.packaging.version.Version` instead of `packaging.version.Version` and seems on some version it doesn't support `.major/minor` operation. changing it back to using `packaging.version.parse`